Fix typos

rex4539 opened the pull request.

syohex commented on src/_cmake:
```diff
@@ -68,10 +68,10 @@ local cmake_build_options;cmake_build_options=(
     '--warn-uninitialized[Warn about uninitialized values.]'
     '--warn-unused-vars[Warn about unused variables.]'
-    '--no-warn-unused-cli[Dont warn about command line options.]'
+    '--no-warn-unused-cli[Don't warn about command line options.]'
```
This causes parser error. You need to use `'\''` 

```
_cmake:125: parse error near `}'
```

syohex commented on src/_drush:
```diff
@@ -51,7 +51,7 @@ _drush() {
     '(- *)'{-y,--yes}'[Assume "yes" as answer to all prompts.]' \
     '(- *)'{-n,--no}'[Assume "no" as answer to all prompts.]' \
-    '(- *)'{-s,--simulate}'[Simulate all relevant actions (dont actually change the system).]' \
+    '(- *)'{-s,--simulate}'[Simulate all relevant actions (don't actually change the system).]' \
```
ditto

syohex commented on src/_drush:
```diff
@@ -114,7 +114,7 @@ case $state in
           '(--destination)--destination=[The full path and filename in which the archive should be stored. If omitted, it will be saved to the drush-backups directory.]' \
           '(--no-core)--no-core[Exclude Drupal core, so the backup only contains the site specific stuff.]' \
-          '(--pipe)--pipe[Only print the destination of the archive. Useful for scripts that dont pass --destination.]' \
+          '(--pipe)--pipe[Only print the destination of the archive. Useful for scripts that don't pass --destination.]' \
```
ditto

syohex commented on src/_drush:
```diff
@@ -171,7 +171,7 @@ case $state in
           {-y,--yes}'[Assume "yes" as answer to all prompts.]' \
           {-n,--no}'[Assume "no" as answer to all prompts.]' \
-          {-s,--simulate}'[Simulate all relevant actions (dont actually change the system).]' \
+          {-s,--simulate}'[Simulate all relevant actions (don't actually change the system).]' \
```
ditto

syohex commented on src/_fab:
```diff
@@ -75,8 +75,8 @@ _arguments -w -S -C \
-  '(-a --no-agent)'{-a,--no-agent}'[don''t use the running SSH agent]' \
-  '(-k --no-keys)'{-k,--no-keys}'[don''t load private key files from ~/.ssh/]' \
+  '(-a --no-agent)'{-a,--no-agent}'[don't use the running SSH agent]' \
+  '(-k --no-keys)'{-k,--no-keys}'[don't load private key files from ~/.ssh/]' \
```
ditto

syohex commented on src/_git-flow:
```diff
@@ -137,7 +137,7 @@ __git-flow-release ()
 						-u'[Use the given GPG-key for the digital signature (implies -s)]'\
 						-m'[Use the given tag message]'\
-						-n'[Dont tag this release ]'\
+						-n'[Don't tag this release]'\
```
ditto

syohex commented on src/_pgsql_utils:
```diff
@@ -204,14 +204,14 @@ _psql () {
         {-s,--single-step}'[prompt before each query]' \
         {-S,--single-line}'[newline sends query]' \
-        {-t,--tuples-only}'[dont display header/footer]' \
+        {-t,--tuples-only}'[don't display header/footer]' \
```
ditto

syohex commented on src/_pgsql_utils:
```diff
@@ -204,14 +204,14 @@ _psql () {
         {-z,--field-separator-zero}'[set field separator for unaligned output to zero byte]' \
         {-0,--record-separator-zero}'[set record separator for unaligned output to zero byte]' \
-        {-X,--no-psqlrc}'[dont read ~/.psqlrc]' \
+        {-X,--no-psqlrc}'[don't read ~/.psqlrc]' \
```
ditto

syohex commented on src/_pgsql_utils:
```diff
@@ -236,15 +236,15 @@ _pg_dump () {
         {-o,--oids}'[dump objects identifiers for every table]' \
-        {-O,--no-owner}'[dont recreate as same owner]' \
-        {-R,--no-reconnect}'[dont output connect]' \
+        {-O,--no-owner}'[don't recreate as same owner]' \
+        {-R,--no-reconnect}'[don't output connect]' \
```
ditto

syohex commented on src/_pgsql_utils:
```diff
@@ -236,15 +236,15 @@ _pg_dump () {
         {-v,--verbose}'[verbose mode]' \
         {-V,--version}'[display client version]' \
-        {-x,--no-{acl,privileges}}'[dont dump ACLs]' \
+        {-x,--no-{acl,privileges}}'[don't dump ACLs]' \
```
ditto

syohex commented on src/_pgsql_utils:
```diff
@@ -328,12 +328,12 @@ _pg_dumpall () {
         {-f+,--file=}':output file:_files' \
         {-o,--oids}'[dump objects identifiers for every table]' \
-        {-O,--no-owner}'[dont recreate as same owner]' \
+        {-O,--no-owner}'[don't recreate as same owner]' \
```
ditto

syohex commented on src/_pgsql_utils:
```diff
@@ -328,12 +328,12 @@ _pg_dumpall () {
         {-S+,--superuser=}':superuser name:_pgsql_users' \
         {-t,--tablespaces-only}'[no databases or roles, only tablespaces]' \
-        {-x,--no-privileges}'[dont dump ACLs]' \
+        {-x,--no-privileges}'[don't dump ACLs]' \
```
ditto

syohex commented on src/_showoff:
```diff
@@ -67,20 +67,20 @@ case $state in
 				_arguments \
 					'(-d --slidedir)'{-d,--slidedir}'=[Sample slide directory name (default: one)]:arg' \
-					'(-n --nosamples)'{-n,--nosamples}'=[Dont create sample slides]' \
+					'(-n --nosamples)'{-n,--nosamples}'=[Don't create sample slides]' \
```
ditto

syohex commented on src/_showoff:
```diff
@@ -67,20 +67,20 @@ case $state in
 					{-s,--source}'=[Include code from the given file as the slide body]:file:_files' \
 					'(-t --style --type)'{-t,--style,--type}'=[Slide Type/Style (default: title)]:style' \
-					'(-u --no-number)'{-u,--no-number}'[Dont number the slide, use the given name verbatim]' \
+					'(-u --no-number)'{-u,--no-number}'[Don't number the slide, use the given name verbatim]' \
```
ditto

syohex commented on src/_cmake:
```diff
@@ -68,10 +68,10 @@ local cmake_build_options;cmake_build_options=(
     '-i[Run in wizard mode]'
     '-L-[List cache variables]::_values "options" "[non-advanced cache variables]" "A[advanced cache variables]" "H[non-advanced cached variables with help]" "AH[advanced cache variables with help]"'
-    '--trace[Put cmake in trace mode]'
+    '--trace[Put CMake in trace mode]'
```
Original code use lowercase `cmake` for this option. Please don't capitalize this `cmake`

rex4539 commented:
Is it ok now?

syohex commented:
> Is it ok now?

No

`'`(single quote) cannot be escaped in single quotes. `'Don\'t'` is invalid syntax in shell script.
It should be `'\''` as I said like `'Don'\''t'`. This consists of three characters, `'` and `\'` and `'`

- `'`  closes preceding string
- put `\'`(single quote)
- `'` starts new string

You can check whether syntax is valid by `source completion_file_path`, for example `source _cmake`.

rex4539 commented:
One more attempt :)

syohex commented on src/_fab:
```diff
@@ -63,7 +63,7 @@ output_levels=(
 _arguments -w -S -C \
   '(-)'{-h,--help}'[show this help message and exit]: :->noargs' \
-  '(-)'{-V,--version}'[show program''s version number and exit]: :->noargs' \
+  '(-)'{-V,--version}'[show program's version number and exit]: :->noargs' \
```
This line is not fixed yet.

syohex commented:
BTW `licence` is not typo, it is British English. I suppose it should not be fixed, keep original word.

rex4539 commented:
I fixed `_fab` but I think the license change is correct, even just for consistency.

I mean the file is named `LICENSE` itself :)

And, for example, in `CONTRIBUTING.md` there are multiple instances where `license` was already written and I just fixed the only instance where is was written `licence`.

Let me know what you think.

okapia commented:
> It should be `'\''` as I said like `'Don'\''t'`. This consists of three characters, `'` and `\'` and `'`

For what it is worth, the usual approach taken in completions distributed in zsh itself is to use double quotes for the whole string when an apostrophe is needed. A single quote inside double quotes doesn't need any additional backslashes. I wouldn't necessarily change any that are working so this is just for information.

MaximDevoir commented:
I think we're okay changing `licence` to `license` so long as the changes are **outside of legal documents or statements**. So switching to `license` within documents like `CONTRIBUTING.md` should be okay.

However, if the word *licence* is used in a license file or is part of some legal text, then we shouldn't change the dialect of the language.

> I mean the file is named LICENSE itself :)

Grammarly has a good article on the difference between [Licence vs. License](https://www.grammarly.com/blog/licence-license/). Being from the US, I've always used *license* in writing. But it sounds like a filename using `license` is the correct usage.

syohex commented:
> I think we're okay changing `licence` to `license` for consistency so long as the changes are **outside of legal documents or statements**. So switching to `license` within documents like `CONTRIBUTING.md` should be okay.
> 
> However, if the word _licence_ is used in a license file or is part of some legal text, then we shouldn't change the dialect of the language.

I agree

MaximDevoir commented on CONTRIBUTING.md:
```diff
@@ -9,7 +9,7 @@ Contributions are welcome, just make sure you follow the guidelines:
  * Please do not just copy/paste someone else's completion, ask before.
  * Partially implemented completions are not accepted.
  * Please add a header containing authors, status and origin of the script and license header if you do not wish to use the Zsh license (example [here](src/_tox)).
- * Any reasonable open source licence is acceptable but note that we recommend the use of the Zsh license and that you should use it if you hope for the function to migrate to zsh itself.
```
With regards to https://github.com/zsh-users/zsh-completions/pull/701#issuecomment-611389469

This is the only instance where `licence` is okay to be `license`. All other instances in this PR should be reverted as the usage is within a legal document or text.

rex4539 commented:
I have reverted the rest of the `licence` instances.

syohex commented on src/_scrub:
```diff
@@ -28,7 +28,7 @@
 # -----------
 #
-#  Completion script for scrub an utility which iteratively writes patterns on files or disk devices to make retrieving the data more difficult. (http://linux.die.net/man/1/scrub).
+#  Completion script for scrubbing a utility which iteratively writes patterns on files or disk devices to make retrieving the data more difficult. (http://linux.die.net/man/1/scrub).
```
I suppose this `scrub` is command name

rex4539 replied:
Good catch! Fixed.
